Fix issue with scaling when it isn't necessary

joshraker opened the pull request:
If the App's state changed but none of the services did, every service was being scaled to the existing container count, size, and profile.

madhuravius commented on aptible/resource_app.go:
```diff
@@ -260,15 +260,15 @@ func scaleServices(d *schema.ResourceData, meta interface{}) error {
 
-	services := d.Get("service").(*schema.Set).List()
-
+	if !d.HasChange("service") {
```
i'd leave a comment in this block as to why this is being done

madhuravius replied:
i think you might want to test this with 3-4 services on one app, and make sure that multiple services (in a list) do not result in a HasChange that's falsely positive if the backend reorders this for any reason (might be wrong or maybe i misunderstand some bits about our backend)

joshraker replied:
I'm not testing or trying to fix the potential issue of reordering right now. Just the glaring issue of scaling all services when none of them have changed.

joshraker commented:
Integration tests are passing.

```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -count 1 -parallel 20 -run TestAccResourceApp -timeout 120m
?       github.com/aptible/terraform-provider-aptible   [no test files]
=== RUN   TestAccResourceApp_basic
--- PASS: TestAccResourceApp_basic (10.87s)
=== RUN   TestAccResourceApp_deploy
Still creating...
Still creating...
Still creating...
Done!
Still creating...
Still creating...
Still creating...
Done!
Still creating...
--- PASS: TestAccResourceApp_deploy (43.74s)
=== RUN   TestAccResourceApp_updateConfig
Still creating...
Still creating...
Still creating...
Done!
Still creating...
Still creating...
Still creating...
Done!
Still creating...
Still creating...
Still creating...
Done!
Still creating...
--- PASS: TestAccResourceApp_updateConfig (61.67s)
=== RUN   TestAccResourceApp_scaleDown
Still creating...
Still creating...
Still creating...
Done!
Still creating...
Still creating...
Still creating...
Done!
Still creating...
Still creating...
Done!
--- PASS: TestAccResourceApp_scaleDown (50.85s)
PASS
ok      github.com/aptible/terraform-provider-aptible/aptible   167.627s
```

I also manually confirmed that the issue has been resolved. I updated the config with the existing provider and it created a `deploy` and `scale` (bottom). Then I changed the config twice after installing the updated provider and it only performed the deploy (top). 

```
+------+-------------------------+----------------+---------------------+-----------+----------+----------------------+
|                                             Operation Report: test-app                                              |
+------+-------------------------+----------------+---------------------+-----------+----------+----------------------+
| ID   | Created At              | Type           | Resource            | Status    | Duration | Detail               |
+------+-------------------------+----------------+---------------------+-----------+----------+----------------------+
| 2896 | 2022-10-27 20:40:03 UTC | app::deploy    | test-app            | succeeded | 0:13     | git:                 |
| 2893 | 2022-10-27 20:39:30 UTC | app::deploy    | test-app            | succeeded | 0:13     | git:                 |
| 2889 | 2022-10-27 20:37:51 UTC | service::scale | test-app-cmd-idczrg | succeeded | 0:09     | count: 1, size: 1024 |
| 2888 | 2022-10-27 20:37:36 UTC | app::deploy    | test-app            | succeeded | 0:12     | git:                 |
+------+-------------------------+----------------+---------------------+-----------+----------+----------------------+
```

madhuravius reviewed (approved):
LGTM with manual test

for now I think this is fine, but we should revisit ordering / guards
